code/util/secrets.hpp

```diff
@@ -0,0 +1,40 @@
+/* Gobby - GTK-based collaborative text editor
+ * Copyright (C) 2024 Philipp Kern
+ *
+ * Permission to use, copy, modify, and/or distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+#ifndef _GOBBY_UTIL_SECRETS_HPP_
+#define _GOBBY_UTIL_SECRETS_HPP_
+
+#include <functional>
+#include <optional>
+#include <string>
+
+namespace Gobby
+{
+
+using LookupCallback = std::function<void(std::optional<std::string> password)>;
```

Comment:
Why not just call it with an empty string if it couldn't be found? Having the C++17 dependency *just* for this seems a bit unjustified.